src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -0,0 +1,1572 @@
+
+                AddMembersByInferredTypeCimType(typename, results, filterToCall);
+            }
+
+            return results;
+        }
+
+        private void AddMembersByInferredTypeCimType(PSTypeName typename, List<object> results, Func<object, bool> filterToCall)
+        {
+            string cimNamespace;
+            string className;
+            if (ParseCimCommandsTypeName(typename, out cimNamespace, out className))
+            {
+                _powerShell.AddCommandWithPreferenceSetting("CimCmdlets\\Get-CimClass")
+                    .AddParameter("Namespace", cimNamespace)
+                    .AddParameter("Class", className);
+                Exception unused;
+                var classes = Helper.ExecuteCurrentPowerShell(out unused);
+                foreach (var cimClass in classes.Select(PSObject.Base).OfType<CimClass>())
+                {
+                    results.AddRange(filterToCall != null
+                        ? cimClass.CimClassProperties.Where(filterToCall)
+                        : cimClass.CimClassProperties);
+                }
+            }
+        }
+
+        private void AddMembersByInferredTypeDefinitionAst(PSTypeName typename, bool isStatic, Func<object, bool> filter, Func<object, bool> filterToCall,
+            List<object> results)
+        {
+            if (CurrentTypeDefinitionAst != typename.TypeDefinitionAst)
+            {
+                if (filterToCall == null)
+                    filterToCall = o => !IsMemberHidden(o);
+                else
+                    filterToCall = o => !IsMemberHidden(o) && filter(o);
+            }
+
+            bool foundConstructor = false;
+            foreach (var member in typename.TypeDefinitionAst.Members)
+            {
+                bool add = false;
+                var propertyMember = member as PropertyMemberAst;
+                if (propertyMember != null)
+                {
+                    add = propertyMember.IsStatic == isStatic;
+                }
+                else
+                {
+                    var functionMember = (FunctionMemberAst) member;
+                    add = functionMember.IsStatic == isStatic;
+                    foundConstructor |= functionMember.IsConstructor;
+                }
+
+                if (filterToCall != null && add)
+                {
+                    add = filterToCall(member);
+                }
+
+                if (add)
+                {
+                    results.Add(member);
+                }
+            }
+
+            //iterate through bases/interfaces
+            foreach (var baseType in typename.TypeDefinitionAst.BaseTypes)
+            {
+                var baseTypeName = baseType.TypeName as TypeName;
+                if (baseTypeName == null) continue;
+                var baseTypeDefinitionAst = baseTypeName._typeDefinitionAst;
+                results.AddRange(GetMembersByInferredType(new PSTypeName(baseTypeDefinitionAst), isStatic, filterToCall));
```

Comment:
`class foo : Hashtable {}`
In this case, `baseType._typeDefinitionAst` is null, and thus `new PSTypeName(baseTypeDefinitionAst)` will throw.